backend/src/apiserver/resource/resource_manager.go

```diff
@@ -219,6 +223,8 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 	if err = workflow.VerifyParameters(parameters); err != nil {
 		return nil, util.Wrap(err, "Failed to verify parameters.")
 	}
+
+	workflow.SetServiceAccount(common.GetStringConfig(defaultPipelineRunnerServiceAccountEnvVar))
```

Comment:
the backend should ignore the value SDK provide. The pipeline-runner service account might not exist in some deployment. E.g. marketplace deployment will use a different service account which name is auto generated.

Reply:
Does it mean that we should remove the service account API from SDK? Is there any use case that user wants to have per-pipeline SA? I don't know any user requested it before. Maybe we can just drop it and always use the server specified one. I am fine with that.

Reply:
IIUC the service account API you talked about is a different thing. That API is for specifying the SA for talking to GCP resources. The SA in this change is KSA which only determine what K8s resource the pipeline could manipulate.

Reply:
> Does it mean that we should remove the service account API from SDK?

I think we should remove it, but only after multiple releases, since this would be a breaking change (The new SDK won't work with old api_server).

>Is there any use case that user wants to have per-pipeline SA?

It might be not the best security practice to allow workflow to control the SAs. Also at this moment the user cannot override the SA using the SDK.

Reply:
I see. You are right. We are not exposing the SA to the DSL user directly today.